### PR TITLE
Enhance preprocess

### PR DESCRIPTION
**Pull request by r0mainK**
3 commits here:

- change default (removed litteral) and no-tokenize option (useless)
- add stop words handling (had to update the DockerFile for nltk data -_-)
- add new stemming logic: reverse mapping to get dominant form, and the 3 rules we'd talked about to remove obvious noise

still building the image but the code should work

### CODE REVIEW

File: tmexp/preprocess.py [hunk 1]
```diff
@@ -255,8 +258,9 @@ def feature_extractor(uast_obj: Any) -> Iterator[Tuple[str, str]]:
         exclude_vendors=exclude_vendors,
         langs=used_langs,
     )
-    if stem:
-        stemmer = PorterStemmer()
+    stop_words = frozenset(stopwords.words("english"))
+    stemmer = PorterStemmer()
+    stem_mapping: Dict[str, WordCount] = defaultdict(lambda: Counter())
```
Comment (m09) [inline tmexp/preprocess.py]:
```suggestion
    stem_mapping: Dict[str, WordCount] = defaultdict(Counter)
```

File: tmexp/preprocess.py [hunk 2]
```diff
@@ -314,20 +318,24 @@ def feature_extractor(uast_obj: Any) -> Iterator[Tuple[str, str]]:
...
+                for word in words
+                for w in re.findall(r"[A-Z]?[a-z]+|[A-Z]+(?=[A-Z]|$)", word)
+            ]
+            words = [w.lower() for w in words]
+            stems_words: List[Tuple[str, str]] = [(stemmer.stem(w), w) for w in words]
+            stems_words = [(s, w) for s, w in stems_words if good_token(s)]
+            if stems_words:
                 num_nodes += 1
-                feature_dict[feature].update(words)
+                feature_dict[feature].update([s for s, _ in stems_words])
```
Comment (m09) [inline tmexp/preprocess.py]:
```suggestion
                feature_dict[feature].update(s for s, _ in stems_words)
```

**Review by m09 — CHANGES_REQUESTED** [review]
Small nitpicks but otherwise :100:

File: tmexp/preprocess.py (hunk 1 above)
Comment (r0mainK) [reply tmexp/preprocess.py]: done

File: tmexp/preprocess.py (hunk 2 above)
Comment (r0mainK) [reply tmexp/preprocess.py]: done

**Review by m09 — APPROVED** [review]
:+1:

**Pull request merged**